src/api/spec/controllers/webui/package_controller_spec.rb

```diff
@@ -2,20 +2,89 @@
 
 RSpec.describe Webui::PackageController, vcr: true do
   let(:admin) { create(:admin_user, login: "admin") }
-  let(:user) { create(:user, login: 'tom') }
+  let(:user) { create(:confirmed_user, login: 'tom') }
   let(:source_project) { user.home_project }
   let(:source_package) { create(:package, name: 'my_package', project: source_project) }
   let(:target_project) { create(:project) }
+  let(:package) { create(:package_with_file, name: "package_with_file", project: source_project) }
 
   describe 'submit_request' do
+    context "sending a valid submit request" do
+      before do
+        login(user)
+        post :submit_request, { project: source_project, package: package, targetproject: target_project }
+      end
+
+      it { expect(flash[:notice]).to match("Created .+submit request \\d.+to .+#{target_project}") }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: package)) }
+
+      it "creates a submit request" do
+        expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: package.name)).to exist
+      end
+    end
+
+    context "having whitespaces in parameters" do
```

Comment:
Would be better to have a shared example called "submit request succesful created" and set the params in each case from outside?